gtk4/src/subclass/orientable.rs

```diff
@@ -0,0 +1,41 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Orientable, Orientation};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+pub trait OrientableImpl: ObjectImpl {
+    fn get_orientation(&self, orientable: &Self::Type) -> Orientation {
+        unsafe {
+            let type_ = ffi::gtk_orientable_get_type();
+            let iface = glib::gobject_ffi::g_type_default_interface_ref(type_)
+                as *mut ffi::GtkOrientableIface;
+            assert!(!iface.is_null());
+
+            let orientation = ffi::gtk_orientable_get_orientation(
+                orientable.unsafe_cast_ref::<Orientable>().to_glib_none().0,
+            );
+
+            glib::gobject_ffi::g_type_default_interface_unref(iface as glib::ffi::gpointer);
```

Comment:
There are no default implementations for any of the functions in `GtkOrientable`

Reply:
Also there are no methods at all in `GtkOrientable` :) It works by having your subclass provide a (appropriately typed) "orientation" property.

The trait should be empty

Reply:
Done.